lib/NormalModule.js

```diff
@@ -125,19 +127,32 @@ const asBuffer = input => {
 	return input;
 };
 
-const readResourceFn = fs => {
-	return (resource, callback) => {
-		const decodedData = decodeDataURI(resource);
-
-		if (decodedData) {
-			process.nextTick(() => {
-				callback(null, decodedData);
-			});
-			return;
+/**
+ * @param {InputFileSystem} fs fs
+ * @param {string|undefined} scheme scheme
+ * @returns {(resource: string, cb: (err: Error|null, data?: string|Buffer) => void) => void} read function
+ */
+const readResourceFn = (fs, scheme) => {
+	if (scheme) {
+		switch (scheme) {
+			case "data":
+				return (resource, callback) => {
+					process.nextTick(() => {
+						callback(null, decodeDataURI(resource));
+					});
+				};
+			case "file":
+				return (resource, callback) => {
+					//@ts-expect-error TODO add URL path support to CachedInputFileSystem
+					fs.readFile(new URL(resource), callback);
```

Comment:
Use https://nodejs.org/api/url.html#url_url_fileurltopath_url to convert it back to a path.